impeller/entity/geometry/cover_geometry.h

```diff
@@ -36,8 +36,6 @@ class CoverGeometry final : public Geometry {
   CoverGeometry& operator=(const CoverGeometry&) = delete;
 };
 
-static_assert(std::is_trivially_destructible<CoverGeometry>::value);
```

Comment:
A bit of a weird assertion to have in the first place TBH. Do we know why this was important? And presumably isn't now?

Reply:
I added this when we were prototyping doing a placement new allocator, but that is made obsolete by the DisplayList alignment